File: data/multilang_comparison/test_zig/src/func22.zig
const lib = @import("lib.zig");
const std = lib.std;
const allocator = lib.allocator;
const Array = lib.Array;
const ArrayParam = lib.ArrayParam;
const get_path = lib.get_path;

const func0 = @import("func0.zig").func0;
const func10 = @import("func10.zig").func10;
const func11 = @import("func11.zig").func11;
const func28 = @import("func28.zig").func28;
const func29 = @import("func29.zig").func29;
const func46 = @import("func46.zig").func46;
const func47 = @import("func47.zig").func47;
const func58 = @import("func58.zig").func58;
const func59 = @import("func59.zig").func59;
const func48 = @import("func48.zig").func48;
const func60 = @import("func60.zig").func60;
const func61 = @import("func61.zig").func61;
const func62 = @import("func62.zig").func62;
const func63 = @import("func63.zig").func63;
const func52 = @import("func52.zig").func52;
const func49 = @import("func49.zig").func49;
const func53 = @import("func53.zig").func53;
const func40 = @import("func40.zig").func40;
const func41 = @import("func41.zig").func41;
const func54 = @import("func54.zig").func54;
const func55 = @import("func55.zig").func55;
const func56 = @import("func56.zig").func56;
const func57 = @import("func57.zig").func57;
const func50 = @import("func50.zig").func50;
const func30 = @import("func30.zig").func30;
const func42 = @import("func42.zig").func42;
const func43 = @import("func43.zig").func43;
const func44 = @import("func44.zig").func44;
const func45 = @import("func45.zig").func45;
const func51 = @import("func51.zig").func51;
const func34 = @import("func34.zig").func34;
const func31 = @import("func31.zig").func31;
const func35 = @import("func35.zig").func35;
const func23 = @import("func23.zig").func23;
const func36 = @import("func36.zig").func36;
const func37 = @import("func37.zig").func37;
const func38 = @import("func38.zig").func38;
const func39 = @import("func39.zig").func39;
const func32 = @import("func32.zig").func32;
const func12 = @import("func12.zig").func12;
const func24 = @import("func24.zig").func24;
const func25 = @import("func25.zig").func25;
const func26 = @import("func26.zig").func26;
const func27 = @import("func27.zig").func27;
const func33 = @import("func33.zig").func33;
const func16 = @import("func16.zig").func16;
const func13 = @import("func13.zig").func13;
const func17 = @import("func17.zig").func17;
const func4 = @import("func4.zig").func4;
const func5 = @import("func5.zig").func5;
const func18 = @import("func18.zig").func18;
const func19 = @import("func19.zig").func19;
const func20 = @import("func20.zig").func20;
const func21 = @import("func21.zig").func21;
const func14 = @import("func14.zig").func14;
const func6 = @import("func6.zig").func6;
const func7 = @import("func7.zig").func7;
const func8 = @import("func8.zig").func8;
const func9 = @import("func9.zig").func9;
const func15 = @import("func15.zig").func15;
const func2 = @import("func2.zig").func2;
const func1 = @import("func1.zig").func1;
const func3 = @import("func3.zig").func3;

pub fn func22(vars: *ArrayParam, _path0: u64) !*Array {
   var pCounter: usize = vars.size;
   _ = &pCounter;
   const path0 = _path0;
   _ = &path0;
   var array113: *Array = undefined;
   if (pCounter > 0) {
       pCounter -= 1;
       array113 = vars.data[pCounter];
       array113.refC += 1;
   } else {
       array113 = try allocator.create(Array);
       array113.*.size = 850;
       array113.*.refC = 1;
       array113.*.id = 113;
       array113.*.data = try allocator.alloc(u32, array113.*.size);
   }
   var loop35: usize = 0;
   const loopLimit35: usize = (50)/4 + 1;
   while (loop35 < loopLimit35) {
      if ((path0 & 1) != 0) {
         var array114: *Array = undefined;
         if (pCounter > 0) {
             pCounter -= 1;
             array114 = vars.data[pCounter];
             array114.refC += 1;
         } else {
             array114 = try allocator.create(Array);
             array114.*.size = 255;
             array114.*.refC = 1;
             array114.*.id = 114;
             array114.*.data = try allocator.alloc(u32, array114.*.size);
         }
         var loop36: usize = 0;
         const loopLimit36: usize = (50)/5 + 1;
         while (loop36 < loopLimit36) {
            for (0..array113.size) |i| {
                array113.data[i] -= 1;
            }
            for (0..array113.size) |i| {
                if (array113.data[i] == 79) {
                    return array113;
                }
            }
            loop36 += 1;
         }
         const params0_data = try allocator.alloc(*Array, 2);
         params0_data[0] = array113;
         params0_data[1] = array114;
         var params0 = ArrayParam{
             .data = params0_data,
             .size = 2,
         };
         const array115 = try func42(&params0, try get_path());
         defer allocator.free(params0.data);
         array115.refC -= 1;
         if (array115.refC == 0) {
             allocator.free(array115.data);
             allocator.destroy(array115);
         }
         array114.refC -= 1;
         if (array114.refC == 0) {
             allocator.free(array114.data);
             allocator.destroy(array114);
         }
      
      } else {
         const params0_data = try allocator.alloc(*Array, 1);
         params0_data[0] = array113;
         var params0 = ArrayParam{
             .data = params0_data,
             .size = 1,
         };
         const array116 = try func34(&params0, try get_path());
         defer allocator.free(params0.data);
         array116.refC -= 1;
         if (array116.refC == 0) {
             allocator.free(array116.data);
             allocator.destroy(array116);
         }
      }
      var array117: *Array = undefined;
      if (pCounter > 0) {
          pCounter -= 1;
          array117 = vars.data[pCounter];
          array117.refC += 1;
      } else {
          array117 = try allocator.create(Array);
          array117.*.size = 567;
          array117.*.refC = 1;
          array117.*.id = 117;
          array117.*.data = try allocator.alloc(u32, array117.*.size);
      }
      const params0_data = try allocator.alloc(*Array, 2);
      params0_data[0] = array113;
      params0_data[1] = array117;
      var params0 = ArrayParam{
          .data = params0_data,
          .size = 2,
      };
      const array118 = try func35(&params0);
      defer allocator.free(params0.data);
      if ((path0 & 2) != 0) {
         var array119: *Array = undefined;
         if (pCounter > 0) {
             pCounter -= 1;
             array119 = vars.data[pCounter];
             array119.refC += 1;
         } else {
             array119 = try allocator.create(Array);
             array119.*.size = 961;
             array119.*.refC = 1;
             array119.*.id = 119;
             array119.*.data = try allocator.alloc(u32, array119.*.size);
         }
         var loop37: usize = 0;
         const loopLimit37: usize = (50)/5 + 1;
         while (loop37 < loopLimit37) {
            for (0..array118.size) |i| {
                array118.data[i] -= 1;
            }
            for (0..array118.size) |i| {
                if (array118.data[i] == 59) {
                    return array118;
                }
            }
            loop37 += 1;
         }
         const params1_data = try allocator.alloc(*Array, 4);
         params1_data[0] = array113;
         params1_data[1] = array117;
         params1_data[2] = array118;
         params1_data[3] = array119;
         var params1 = ArrayParam{
             .data = params1_data,
             .size = 4,
         };
         const array120 = try func46(&params1, try get_path());
         defer allocator.free(params1.data);
         array120.refC -= 1;
         if (array120.refC == 0) {
             allocator.free(array120.data);
             allocator.destroy(array120);
         }
         array119.refC -= 1;
         if (array119.refC == 0) {
             allocator.free(array119.data);
             allocator.destroy(array119);
         }
      
      } else {
         const params1_data = try allocator.alloc(*Array, 3);
         params1_data[0] = array113;
         params1_data[1] = array117;
         params1_data[2] = array118;
         var params1 = ArrayParam{
             .data = params1_data,
             .size = 3,
         };
         const array121 = try func40(&params1, try get_path());
         defer allocator.free(params1.data);
         array121.refC -= 1;
         if (array121.refC == 0) {
             allocator.free(array121.data);
             allocator.destroy(array121);
         }
      }
      for (0..array118.size) |i| {
          if (array118.data[i] == 21) {
              return array118;
          }
      }
      array118.refC -= 1;
      if (array118.refC == 0) {
          allocator.free(array118.data);
          allocator.destroy(array118);
      }
      array117.refC -= 1;
      if (array117.refC == 0) {
          allocator.free(array117.data);
          allocator.destroy(array117);
      }
      loop35 += 1;
   }
   for (0..array113.size) |i| {
       array113.data[i] -= 1;
   }
   for (0..array113.size) |i| {
       if (array113.data[i] == 28) {
           return array113;
       }
   }
   return array113;
}

File: data/multilang_comparison/program_nim/src/func22.nim
proc func22*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array113: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array113 = vars.data[pCounter]
      inc array113.refC
      debugCopy(array113.id)
   else:
      array113 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array113.size = 179
      array113.refC = 1
      array113.id = 113
      array113.data = cast[ptr UncheckedArray[uint32]](alloc(array113.size * sizeof(uint32)))
      zeroMem(array113.data, array113.size * sizeof(uint32))
      debugNew(array113.id)
   let loopLimit35 = (50) div (4) + 1
   for loop35 in 0 ..< loopLimit35:
      discard
      if (PATH0 and 1'u64) != 0:
         discard
         var array114: ptr ArrayT
         if pCounter > 0:
            dec pCounter
            array114 = vars.data[pCounter]
            inc array114.refC
            debugCopy(array114.id)
         else:
            array114 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
            array114.size = 418
            array114.refC = 1
            array114.id = 114
            array114.data = cast[ptr UncheckedArray[uint32]](alloc(array114.size * sizeof(uint32)))
            zeroMem(array114.data, array114.size * sizeof(uint32))
            debugNew(array114.id)
         let loopLimit36 = (50) div (5) + 1
         for loop36 in 0 ..< loopLimit36:
            discard
            for i in 0 ..< array114.size:
               dec array114.data[i]
            for i in 0 ..< array113.size:
               if array113.data[i] == 48:
                  return array113
         var params0: ArrayTParam
         params0.size = 2
         params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
         params0.data[0] = array113
         params0.data[1] = array114
         var array115: ptr ArrayT = func42(addr params0, getPath(), loopsFactor)
         debugReturn(array115.id)
         dealloc(params0.data)
         dec array115.refC
         if array115.refC == 0:
            dealloc(array115.data)
            dealloc(array115)
            debugFree(array115.id)
         dec array114.refC
         if array114.refC == 0:
            dealloc(array114.data)
            dealloc(array114)
            debugFree(array114.id)
      else:
         discard
         var params0: ArrayTParam
         params0.size = 1
         params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
         params0.data[0] = array113
         var array116: ptr ArrayT = func34(addr params0, getPath(), loopsFactor)
         debugReturn(array116.id)
         dealloc(params0.data)
         dec array116.refC
         if array116.refC == 0:
            dealloc(array116.data)
            dealloc(array116)
            debugFree(array116.id)
      var array117: ptr ArrayT
      if pCounter > 0:
         dec pCounter
         array117 = vars.data[pCounter]
         inc array117.refC
         debugCopy(array117.id)
      else:
         array117 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
         array117.size = 210
         array117.refC = 1
         array117.id = 117
         array117.data = cast[ptr UncheckedArray[uint32]](alloc(array117.size * sizeof(uint32)))
         zeroMem(array117.data, array117.size * sizeof(uint32))
         debugNew(array117.id)
      var params0: ArrayTParam
      params0.size = 2
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      params0.data[0] = array113
      params0.data[1] = array117
      var array118: ptr ArrayT = func35(addr params0, loopsFactor)
      debugReturn(array118.id)
      dealloc(params0.data)
      if (PATH0 and 2'u64) != 0:
         discard
         var array119: ptr ArrayT
         if pCounter > 0:
            dec pCounter
            array119 = vars.data[pCounter]
            inc array119.refC
            debugCopy(array119.id)
         else:
            array119 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
            array119.size = 206
            array119.refC = 1
            array119.id = 119
            array119.data = cast[ptr UncheckedArray[uint32]](alloc(array119.size * sizeof(uint32)))
            zeroMem(array119.data, array119.size * sizeof(uint32))
            debugNew(array119.id)
         let loopLimit37 = (50) div (5) + 1
         for loop37 in 0 ..< loopLimit37:
            discard
            for i in 0 ..< array117.size:
               dec array117.data[i]
            for i in 0 ..< array117.size:
               if array117.data[i] == 72:
                  return array117
         var params1: ArrayTParam
         params1.size = 4
         params1.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params1.size * sizeof(ptr ArrayT)))
         params1.data[0] = array113
         params1.data[1] = array117
         params1.data[2] = array118
         params1.data[3] = array119
         var array120: ptr ArrayT = func46(addr params1, getPath(), loopsFactor)
         debugReturn(array120.id)
         dealloc(params1.data)
         dec array120.refC
         if array120.refC == 0:
            dealloc(array120.data)
            dealloc(array120)
            debugFree(array120.id)
         dec array119.refC
         if array119.refC == 0:
            dealloc(array119.data)
            dealloc(array119)
            debugFree(array119.id)
      else:
         discard
         var params1: ArrayTParam
         params1.size = 3
         params1.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params1.size * sizeof(ptr ArrayT)))
         params1.data[0] = array113
         params1.data[1] = array117
         params1.data[2] = array118
         var array121: ptr ArrayT = func40(addr params1, getPath(), loopsFactor)
         debugReturn(array121.id)
         dealloc(params1.data)
         dec array121.refC
         if array121.refC == 0:
            dealloc(array121.data)
            dealloc(array121)
            debugFree(array121.id)
      for i in 0 ..< array113.size:
         if array113.data[i] == 4:
            return array113
      dec array118.refC
      if array118.refC == 0:
         dealloc(array118.data)
         dealloc(array118)
         debugFree(array118.id)
      dec array117.refC
      if array117.refC == 0:
         dealloc(array117.data)
         dealloc(array117)
         debugFree(array117.id)
   for i in 0 ..< array113.size:
      dec array113.data[i]
   for i in 0 ..< array113.size:
      if array113.data[i] == 40:
         return array113
   return array113

